Framework/Source/Graphics/RenderGraph/RenderGraph.cpp

```diff
@@ -520,36 +521,36 @@ namespace Falcor
         }
     }
 
-    bool canFieldsConnect(const RenderPassReflection::Field& src, const RenderPassReflection::Field& dest)
+    bool canFieldsConnect(const RenderPassReflection::Field& src, const RenderPassReflection::Field& dst)
     {
-        assert(is_set(src.getType(), RenderPassReflection::Field::Type::Output) && is_set(dest.getType(), RenderPassReflection::Field::Type::Input));
+        assert(is_set(src.getType(), RenderPassReflection::Field::Type::Output) && is_set(dst.getType(), RenderPassReflection::Field::Type::Input));
         
-        return (src.getFormat() == dest.getFormat() || dest.getFormat() == ResourceFormat::Unknown) &&
-            src.getResourceType() == dest.getResourceType() &&
-            src.getSampleCount() == dest.getSampleCount();
+        return (src.getFormat() == dst.getFormat() || dst.getFormat() == ResourceFormat::Unknown) &&
```

Comment:
This thing is confusing, and we are going to add more checks. Please refactor it to be on multiple lines (single check with `return false` if the check fails)